src/api/jacobian.jl

```diff
@@ -43,21 +43,18 @@ function jacobian{A}(f, ::Type{A}=Void;
                      chunk_size::Int=default_chunk_size,
                      cache::ForwardDiffCache=ForwardDiffCache(),
                      output_length::Int=0)
-    # if output_length > 0, assume that f is of 
-    # the form f!(output, x), and generate the 
+    # if output_length > 0, assume that f is of
+    # the form f!(output, x), and generate the
     # appropriate closure
     if output_length > 0
-        newf = @eval begin
-            output_cache = ForwardDiffCache()
-            @inline function newf{N,T,C}(x::Vector{GradientNumber{N,T,C}})
-                output = get_workvec!(output_cache, 
-                                      GradientNumber, T,
-                                      Val{$output_length}, 
-                                      Val{N})
-                $f(output, x)
-                return output
-            end
-            return newf
+        output_cache = ForwardDiffCache()
+        @inline function newf{N,T,C}(x::Vector{GradientNumber{N,T,C}})
```

Comment:
My assumption was that inner functions are not always inlined by default (I don't see why they should be), and that `@inline` would force inlining when `newf` gets called [here](https://github.com/JuliaDiff/ForwardDiff.jl/blob/rmeval/src/api/jacobian.jl#L104) and [here](https://github.com/JuliaDiff/ForwardDiff.jl/blob/rmeval/src/api/jacobian.jl#L149).

The reason I want this inlined is because the whole `newf` thing was simply a mechanism to easily support the `f!(output, x)` style without having to write a bunch of new code (a solution to #63 is probably going to involve some new code anyway, though). From that perspective, `newf` really is a "just-insert-some-code-at-the-call-site" kind of function (or at least it seemed that way when I did the original performance tweaking on this implementation).